extensions/script-libraries/mime/mime.livecodescript

```diff
@@ -0,0 +1,1558 @@
+            break 
+         case (tByteNum is 9 or tByteNum is 32) and  tLineLength is not 76 and tIndex is not tNumBytes
+         case tByteNum >= 33 and tByteNum <= 126 and tByteNum is not 61
+            put tByte after tData
+            add 1 to tLineLength
+            break
+         default
+            if tLineLength >= 74 and \
+                  not (tLineLength is 74 and tIndex is tNumBytes) then
+               put "=" & crlf after tData
+               put 3 into tLineLength
+            end if
+            
+            local tConverted
+            put baseConvert(tByteNum,10,16) into tConverted
+            if the length of tConverted is 1 then
+               put 0 before tConverted
+            end if
+            put "=" & tConverted after tData
+            add 3 to tLineLength
+            break
+      end switch
+      
+      put tByteNum into tPrevByteNum
+      add 1 to tIndex
+   end repeat
+   
+   put tData into xData
+end __EncodeQuotedPrintable
+
+/**
+
+Build a multipart mime content type from an array of pre-encoded content
+
+Example:
+put "text/html" into tParamA["type"]
+MIMECreateMultipartContent tContentA, "related", tParamA
+put it into tMailBody
+
+Description:
+This command takes an array of pre-encoded content and creates a multipart
+content type header and boundary to build a single multipart encoded value.
+
+Parameters:
+pContentA (array):
+A numerically indexed array of pre-encoded parts including headers. If no headers are present
+a leading crlf should be present.
+
+pMultipartType (optional string):
+The type of multipart content. If not provided the default type will be `mixed`.
+
+pParamA (optional array):
+Additional parameter array in key : value pairs
+
+Returns:
+The multipart encoded content
+
+*/
+
+command MIMECreateMultipartContent pContentA, pMultipartType, pParamA
+   if pMultipartType is empty then
+      put "mixed" into pMultipartType
+   end if
+   
+   local tBoundary
+   put "Boundary-" & uuid() into tBoundary
+   
+   local tData
+   put "Content-Type: multipart/" & pMultipartType into tData
+   
+   local tParam
+   repeat for each key tParam in pParamA
+      put ";" & crlf & tab & tParam & "=" & quote & pParamA[tParam] & quote after tData
+   end repeat
+   
+   put ";" & crlf & tab & "boundary=" & tBoundary & crlf & crlf after tData
+   
+   local tIndex
+   repeat with tIndex = 1 to the number of elements of pContentA
+      put crlf & "--" & tBoundary & crlf & pContentA[tIndex] after tData
+   end repeat
+   
+   put crlf & "--" & tBoundary & "--" & crlf & crlf after tData
+   
+   return tData for value
+end MIMECreateMultipartContent
+
+
+/**
+
+Generate multipart content from a LiveCode field
+
+Example:
+MIMECreateMultipartContentFromField the long id of field "email"
+if the result is empty then
+   SendEmail it
+end if
+
+Description:
+Parses the content of a field to create multipart conten
+
+Parameters:
+pField (string):
+Field object long id
```

Comment:
Subjective nitpick: I think this might read more easily as "long id of the field to encode" or similar (especially as the other parameter descriptions in the PR are more prose-like).